server/src-lib/Hasura/GraphQL/Execute/LiveQuery/Poll.hs

```diff
@@ -350,21 +457,25 @@ pollQuery metrics batchSize pgExecCtx pgQuery handler =
       let cohortSnapshot = CohortSnapshot cohortVars respRef (map snd curOpsL) (map snd newOpsL)
       return (resId, cohortSnapshot)
 
-    getQueryVars cohortSnapshotMap =
-      Map.toList $ fmap _csVariables cohortSnapshotMap
-
-    getCohortOperations cohortSnapshotMap actionMeta = \case
+    getCohortOperations cohorts = \case
       Left e ->
         -- TODO: this is internal error
         let resp = GQExecError [encodeGQLErr False e]
-        in [ (resp, Nothing, actionMeta, snapshot)
-           | (_, snapshot) <- Map.toList cohortSnapshotMap
-           ]
-      Right responses ->
-        flip mapMaybe responses $ \(respId, result) ->
+        in [(resp, cohortId, Nothing, snapshot) | (cohortId, snapshot) <- cohorts]
+      Right responses -> do
+        let cohortSnapshotMap = Map.fromList cohorts
+        flip mapMaybe responses $ \(cohortId, result) ->
           -- TODO: change it to use bytestrings directly
-          let -- No reason to use lazy bytestrings here, since (1) we fetch the entire result set
-              -- from Postgres strictly and (2) even if we didn’t, hashing will have to force the
-              -- whole thing anyway.
-              respHash = mkRespHash (encJToBS result)
-          in (GQSuccess result, Just $! respHash, actionMeta,) <$> Map.lookup respId cohortSnapshotMap
+          let -- No reason to use lazy bytestrings here, since (1) we fetch the
+              -- entire result set from Postgres strictly and (2) even if we
+              -- didn’t, hashing will have to force the whole thing anyway.
+              respBS = encJToBS result
+              respHash = mkRespHash respBS
+              respSize = BS.length respBS
+          -- TODO: currently we ignore the cases when the cohortId from
+          -- Postgres response is not present in the cohort map of this batch
+          -- (this shouldn't happen but if it happens it means a logic error and
+          -- we should log it)
```

Comment:
We should probably just do this rather than leave a TODO comment, right? It doesn’t seem challenging.